shared/StringCommon.F90

```diff
@@ -0,0 +1,219 @@
+module mapl3g_StringCommon
+   implicit none(type, external)
+   private
+
+   public :: ASCII_UPPER_SHIFT
+   public :: to_string
+   public :: to_character_array
+   public :: is_lower
+   public :: is_upper
+   public :: is_alpha
+   public :: is_digit
+   public :: is_alphanum_character
+   public :: is_alpha_only
+   public :: is_numeric
+   public :: is_alphanumeric
+   public :: lowercase
+   public :: uppercase
+   public :: to_lower
+   public :: to_upper
+   public :: capitalize
+   public :: get_ascii_range
+
+   interface to_lower
+      module procedure :: to_lower_string
+   end interface to_lower
+
+   interface to_upper
+      module procedure :: to_upper_string
+   end interface to_upper
+
+   interface capitalize
+      module procedure :: capitalize_string
+   end interface capitalize
+
+   interface get_ascii_range
+      module procedure :: get_ascii_range_array
+      module procedure :: get_ascii_range_string
+   end interface get_ascii_range
+
+   ! ASCII_UPPER_SHIFT = iachar('A')-iachar('a')
+   integer, parameter :: ASCII_UPPER_SHIFT = -32
```

Comment:
Why can't you use the commented line?

Reply:
I could. Since it is defined by the ASCII standard, I made it a literal constant. I don't think it will make a difference functionally. If you think it is more clear for human eyes. I'm fine with changing it.

Reply:
I changed it on further consideration.